File: 0001.py
class Solution:
    def twoSum(self, nums, target):
        intmap = dict()
        for i in range(len(nums)):
            n = nums[i]
            if (target - n) in intmap:
                return [intmap[target - n], i]
            else:
                intmap[n] = i

File: 0001.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> idict;
        for(int i = 0; i < nums.size(); ++i) {
            int n = nums[i];
            if(idict.count(target-n) != 0) {
                return { idict[target-n], i };
            }else {
                idict[n] = i;
            }
        }
        return {};
    }
};
